# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../heimdal/emacs-pop.diff
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -urN emacs-22.0.97.orig/lib-src/pop.c emacs-22.0.97/lib-src/pop.c
--- emacs-22.0.97.orig/lib-src/pop.c	2007-03-19 09:36:41.000000000 +1000
+++ emacs-22.0.97/lib-src/pop.c	2007-04-09 10:56:32.000000000 +1000
@@ -1181,16 +1181,16 @@
       krb5_free_principal (kcontext, server);
       if (rem)
 	{
-	  if (err_ret && err_ret->text.length)
+	  if (err_ret && err_ret->e_data->length)
 	    {
 	      strcpy (pop_error, KRB_ERROR);
 	      strncat (pop_error, error_message (rem),
 		       ERROR_MAX - sizeof (KRB_ERROR));
 	      strncat (pop_error, " [server says '",
 		       ERROR_MAX - strlen (pop_error) - 1);
-	      strncat (pop_error, err_ret->text.data,
+	      strncat (pop_error, err_ret->e_data->data,
 		       min (ERROR_MAX - strlen (pop_error) - 1,
-			    err_ret->text.length));
+			    err_ret->e_data->length));
 	      strncat (pop_error, "']",
 		       ERROR_MAX - strlen (pop_error) - 1);
 	    }
diff -urN emacs-22.0.97.orig/src/config.in emacs-22.0.97/src/config.in
--- emacs-22.0.97.orig/src/config.in	2007-01-27 02:33:17.000000000 +1000
+++ emacs-22.0.97/src/config.in	2007-04-09 10:58:18.000000000 +1000
@@ -283,7 +283,7 @@
 #undef HAVE_KERBEROS_KRB_H
 
 /* Define to 1 if you have the <krb5.h> header file. */
-#undef HAVE_KRB5_H
+#define HAVE_KRB5_H
 
 /* Define to 1 if you have the <krb.h> header file. */
 #undef HAVE_KRB_H
@@ -325,7 +325,7 @@
 #undef HAVE_LIBKRB4
 
 /* Define to 1 if you have the `krb5' library (-lkrb5). */
-#undef HAVE_LIBKRB5
+#define HAVE_LIBKRB5
 
 /* Define to 1 if you have the `kstat' library (-lkstat). */
 #undef HAVE_LIBKSTAT
